### PR TITLE
Blob: Consume and return ArrayBufferLike

### PR DESCRIPTION
**Pull request by DLehenbauer**
I like the new blob API and propose we go a step further, passing buffers in both directions.

### CODE REVIEW

File: packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts [hunk 1]
```diff
@@ -177,12 +177,12 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
         return response.content;
     }
 
-    public async readBlob(blobid: string): Promise<IsoBuffer> {
+    public async readBlob(blobId: string): Promise<ArrayBufferLike> {
```
Comment (DLehenbauer) [inline packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts:180]: (Just camel-casing 'blobId')

File: packages/drivers/routerlicious-driver/src/documentStorageService.ts [hunk 2]
```diff
@@ -87,12 +87,16 @@ export class DocumentStorageService implements IDocumentStorageService {
         throw new Error("NOT IMPLEMENTED!");
     }
 
-    public async createBlob(file: Uint8Array): Promise<ICreateBlobResponse> {
-        const response = this.manager.createBlob(Uint8ArrayToString(file, "base64"), "base64");
+    public async createBlob(file: ArrayBufferLike): Promise<ICreateBlobResponse> {
+        const response = this.manager.createBlob(
+            Uint8ArrayToString(
+                new Uint8Array(file), "base64"),
```
Comment (DLehenbauer) [inline packages/drivers/routerlicious-driver/src/documentStorageService.ts:93]: (Note that 'new Uint8Array()' does not copy the underlying data...  although transcoding to base64 certainly does.)

File: packages/drivers/routerlicious-driver/src/documentStorageService.ts [hunk 3]
```diff
@@ -87,13 +87,24 @@ export class DocumentStorageService implements IDocumentStorageService {
...
-        return IsoBuffer.from(await this.read(blobId), "base64");
+    public async readBlob(blobId: string): Promise<ArrayBufferLike> {
+        const iso = IsoBuffer.from(await this.read(blobId), "base64");
+
+        // In a Node environment, IsoBuffer may be a Node.js Buffer.  Node.js will
+        // pool multiple small Buffer instances into a single ArrayBuffer, in which
+        // case we need to slice the appropriate span of bytes.
+        return iso.byteLength === iso.buffer.byteLength
+            ? iso.buffer
+            : iso.buffer.slice(iso.byteOffset, iso.byteOffset + iso.byteLength);
```
Comment (DLehenbauer) [inline packages/drivers/routerlicious-driver/src/documentStorageService.ts:107]: Last I checked, Node may bool buffers less  < 4kb into 8kb ArrayBuffers.

So if we slice, and the underlying slice copies, it'll be a small copy.  I don't feel too bad about that, given we're already allocating to decode base64, but we could potentially avoid the copy if we stopped going through IsoBuffer to decode base64.

**Review by wes-carlson — APPROVED** [review]
Looks good, thanks!

**Review by tanviraumi — APPROVED** [review]

**Pull request merged**